### PR TITLE
Adds estimateloan RPC

### PR DESCRIPTION
**Pull request by Jouzo**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:

Adds estimateloan RPC.
This RPC calculates the amount of loan tokens a vault can take depending on its collaterals and an optional targetRatio parameter.

#### Which issue(s) does this PR fixes?:

Fixes part of #796

### CODE REVIEW

File: test/functional/feature_loan_estimateloan.py [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+# file LICENSE or http://www.opensource.org/licenses/mit-license.php.
+"""Test Loan Scheme."""
+
+from test_framework.test_framework import DefiTestFramework
+
+from test_framework.authproxy import JSONRPCException
+from test_framework.util import assert_equal
+import time
+
+class DepositToVaultTest (DefiTestFramework):
```
Comment (Bushstar) [inline test/functional/feature_loan_estimateloan.py:14]: Very minor. Rename test from DepositToVaultTest, also update the comment "Test Loan Scheme" above which is also not updated on the DepositToVaultTest used as a template.

File: src/masternodes/rpc_vault.cpp [hunk 2]
```diff
@@ -1146,6 +1146,104 @@ UniValue listauctionhistory(const JSONRPCRequest& request) {
...
+            }
+
+            auto priceFeed = pcustomcsview->GetFixedIntervalPrice(loanToken->fixedIntervalPriceId);
+            if (!priceFeed.ok) {
+                throw JSONRPCError(RPC_DATABASE_ERROR, priceFeed.msg);
+            }
+
+            auto price = priceFeed.val->priceRecord[0];
+            if (!priceFeed.val->isLive(pcustomcsview->GetPriceDeviation())) {
+                throw JSONRPCError(RPC_MISC_ERROR, strprintf("No live fixed price for %s", tokenId));
```
Comment (bvbfan) [inline src/masternodes/rpc_vault.cpp:1231]: loanToken->symbol is better fit? At least tokenId.v is int not string.

File: src/masternodes/rpc_vault.cpp [hunk 3]
```diff
@@ -1146,6 +1146,104 @@ UniValue listauctionhistory(const JSONRPCRequest& request) {
...
+
+            auto availableValue = MultiplyAmounts(rate.val->totalCollaterals, split);
+            auto loanAmount = DivideAmounts(availableValue, price);
+            auto amountRatio = MultiplyAmounts(DivideAmounts(loanAmount, ratio), 100);
+
+            loanBalances.Add({token->first, amountRatio});
+            totalSplit += split;
+        }
+        if (totalSplit != COIN)
+            throw JSONRPCError(RPC_MISC_ERROR, strprintf("total split between loan tokens = %d vs expected %d", totalSplit, COIN));
```
Comment (bvbfan) [inline src/masternodes/rpc_vault.cpp:1242]: `GetDecimaleString` to `totalSplit` and `COIN`.

**Review by bvbfan — CHANGES_REQUESTED** [review]

File: src/masternodes/rpc_vault.cpp (hunk 2 above)
Comment (Jouzo) [reply src/masternodes/rpc_vault.cpp:1231]: Here tokenId is string already

File: src/masternodes/rpc_vault.cpp (hunk 3 above)
Comment (Jouzo) [reply src/masternodes/rpc_vault.cpp]: Fixed

**Review by bvbfan — APPROVED** [review]

**Review by Bushstar — APPROVED** [review]

**Review by bvbfan — DISMISSED** [review]

**Comment by prasannavl** [comment]
Could you please resolve the conflict when you get a chance? Thanks.

**Review by bvbfan — APPROVED** [review]

**Pull request merged**